hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java

```diff
@@ -0,0 +1,91 @@
+    //    balance changed in the ending staking period), calculate the reward it has earned in the ending staking
+    //    period as its pending reward
+    // 3. Otherwise, there's no staking metadata or balance change for the entity since the start of the ending staking
+    //    period, add the reward earned in the ending period to the current as the new pending reward
+    String UPDATE_ENTITY_STAKE_SQL = """
+            with ending_period_node_stake as (
+              select node_id, epoch_day, reward_rate
+              from node_stake
+              where consensus_timestamp = (select max(consensus_timestamp) from node_stake)
+            ), proxy_staking as (
+              select staked_account_id, sum(balance) as staked_to_me
+              from entity_state_start
+              where deleted is false and staked_account_id <> 0
+              group by staked_account_id
+            ), updated as (
+              select
+                ess.decline_reward as decline_reward_start,
+                (select epoch_day from ending_period_node_stake limit 1) as end_stake_period,
+                ess.id,
+                (case
+                   when ess.deleted is true
+                        or coalesce(es.decline_reward_start, true) is true
+                        or coalesce(es.staked_node_id_start, -1) = -1
+                        then 0
+                   when ess.stake_period_start >= (select epoch_day from ending_period_node_stake limit 1)
+                        then reward_rate * (es.stake_total_start / 100000000)
+                   else es.pending_reward + reward_rate * (es.stake_total_start / 100000000)
+                  end) as pending_reward,
+                ess.staked_node_id as staked_node_id_start,
+                coalesce(ps.staked_to_me, 0) as staked_to_me,
+                (case when ess.decline_reward is true or ess.staked_node_id = -1 then 0
+                      else ess.balance + coalesce(ps.staked_to_me, 0)
+                  end) as stake_total_start
+              from entity_state_start ess
+              left join entity_stake es on es.id = ess.id
+              left join ending_period_node_stake on node_id = es.staked_node_id_start
+              left join proxy_staking ps on ps.staked_account_id = ess.id
+            )
+            insert into entity_stake
+            table updated
+            on conflict (id) do update
+              set decline_reward_start = excluded.decline_reward_start,
+                  end_stake_period     = excluded.end_stake_period,
+                  pending_reward       = excluded.pending_reward,
+                  staked_node_id_start = excluded.staked_node_id_start,
+                  staked_to_me         = excluded.staked_to_me,
+                  stake_total_start    = excluded.stake_total_start;
+            """;
+
+    @Modifying
+    @Query(value = UPDATE_ENTITY_STAKE_SQL, nativeQuery = true)
```

Comment:
nit: It might be better to put the docs and the SQL inline so docs show up when using the method and so future repository methods to this class don't separate the SQL from the method.
```suggestion
    /**
     * Algorithm to update pending reward: ...
     */
    @Modifying
    @Query(nativeQuery = true, value = """
        with ending_period_node_stake as (...
    """)
```

Reply:
sure